frontend/src/DiagramEditor/DiagramEditor.jsx

```diff
@@ -178,20 +234,97 @@ export class DiagramEditor extends React.Component {
     });
   };
 
-  onComponentMouseDown = (e, selectedComponentId) => {
+  onConnectionMouseDown = (e, connectionId) => {
+    this.setState({
+      selectedConnectionId: connectionId,
+    });
+  };
+
+  onComponentMouseDown = (e, componentId) => {
+    const { isConnecting } = this.state;
+    if (!isConnecting) {
+      this.setState({
+        selectedComponentId: componentId,
+      });
+    }
+
     this.setState({
       isDraggingComponent: true,
-      selectedComponentId,
       previousMouseX: e.clientX,
       previousMouseY: e.clientY,
       initialMouseX: e.clientX,
       initialMouseY: e.clientY,
     });
   };
 
+  onComponentMouseUp = (e, componentId) => {
+    const {
+      isConnecting,
+      selectedComponentId,
+      isDraggingComponent,
+    } = this.state;
+
+    this.setState({ isDraggingComponent: false });
+
+    if (isConnecting) {
+      if (selectedComponentId !== componentId) {
+        this.props.sendChange({
+          operation: "addConnection",
+          data: {
+            from: selectedComponentId,
+            to: componentId,
+            id: `connection_${RandomWords({ exactly: 3, join: "-" })}`,
```

Comment:
why random? why not just make it the ids of the things it's connecting?

Reply:
what's the difference? it's cleaner to have the same logic for generating IDs everywhere: connections, components, groups

Reply:
It would just be a minor optimisation. But, I understand that you want to keep it the same everywhere. In which case it would make sense to abstract it into a function. What do you think?

Reply:
that's a good idea. We can do it in the next PR, though